src/main/java/seedu/address/model/assignment/WorkerRole.java

```diff
@@ -0,0 +1,89 @@
+package seedu.address.model.assignment;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Role;
+
+/**
+ * Represents a Worker being assigned to a Role for a shift in the App.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class WorkerRole {
```

Comment:
Personally, `WorkerRole` as a class name doesn't seem very intuitive. Maybe use `WorkerRoleAssignment` or `WorkerRolePair`. The others can chip in on this as well.